samples/Basic_ControlPoint/app/application.cpp

```diff
@@ -0,0 +1,192 @@
+				Serial.println(_F("render->getVolume(0, Master):"));
+				if(checkResponse(response)) {
+					Serial.print(device.friendlyName());
+					Serial.print(_F(": Current Volume = "));
+					Serial.println(response.getCurrentVolume());
+				}
+				Serial.println();
+			});
+		}
+
+		auto conn = device.getConnectionManager();
+		if(conn != nullptr) {
+			conn->getCurrentConnectionInfo(0, [&device](auto response) {
+				Serial.println();
+				Serial.println(_F("conn->getCurrentConnectionInfo(0):"));
+				if(checkResponse(response)) {
+					Serial.print(device.friendlyName());
+					Serial.println(_F(": Current Connection Info = "));
+					response.printTo(Serial);
+					Serial.println(_F("---"));
+				}
+				Serial.println();
+			});
+		}
+
+		auto transport = device.getAVTransport();
+		if(transport != nullptr) {
+			transport->getDeviceCapabilities(0, [&device](auto response) {
+				Serial.println();
+				Serial.println(_F("transport->getDeviceCapabilities(0):"));
+				if(checkResponse(response)) {
+					Serial.print(device.friendlyName());
+					Serial.println(_F(": Device Capabilities = "));
+					response.printTo(Serial);
+					Serial.println(_F("---"));
+				}
+				Serial.println();
+			});
+		}
+
+		// Keep this device
+		return true;
+	}));
+
+	// Stop after 10 seconds
+	auto timer = new AutoDeleteTimer;
+	timer->initializeMs<10000>(InterruptCallback([]() {
+		controlPoint.cancelSearch();
+		Serial.println();
+		Serial.println(_F("OK, I'm all done for now. What shall we do next?"));
+		Serial.println();
+	}));
+	timer->startOnce();
+}
+
+void findMediaServers()
+{
+	controlPoint.cancelSearch();
+	controlPoint.beginSearch(Delegate<bool(MediaServer1&)>([](auto& device) {
+		// Stop at the first response
+		//		controlPoint.cancelSearch();
+
+		Serial.print(_F("Found: "));
+		Serial.println(device.caption());
+
+		auto dir = device.getContentDirectory();
+		if(dir != nullptr) {
+			auto printBrowseResult = [&device](auto response) {
+				Serial.println();
+				Serial.print(_F("Browse response for "));
+				Serial.print(device.friendlyName());
+				Serial.println(':');
+				if(checkResponse(response)) {
+					XML::Document doc;
+					String s = response.getResult();
+					XML::deserialize(doc, s);
+					XML::serialize(doc, Serial, true);
+					response.printTo(Serial);
+				}
+				Serial.println();
+			};
+
+			dir->browse("0", ContentDirectory1::BrowseFlag::fs_BrowseMetadata, "*", 0, 10, nullptr, printBrowseResult);
+			dir->browse("0", ContentDirectory1::BrowseFlag::fs_BrowseDirectChildren, "*", 0, 10, nullptr,
+						printBrowseResult);
+
+			// Send a bad request, should return a fault condition
+			dir->browse(F("bad robot"), F("chicken"), "", 0, 10, nullptr, printBrowseResult);
+		}
+
+		return true;
+	}));
+
+	// Stop after 5 seconds and do a search for renderers
+	auto timer = new AutoDeleteTimer;
+	timer->initializeMs<5000>(findMediaRenderers).startOnce();
+}
+
+void initUPnP()
+{
+	UPnP::schemas_upnp_org::registerClasses();
```

Comment:
Maybe you can add here comments explaining why is the registerClassess needed and what it is actually doing.